### ISSUE TITLE
Cannot Properly Include Library

### ISSUE THREAD

**Issue by kklouzal** [USCiLab/cereal#346]
Taken from the **Quick Start** page:
>>installed anywhere you can access header files

I've gone ahead and dropped the "cereal" folder into my projects directory (the one which has my .h and .cpp files) then tried to include the necessary headers
```
#include "cereal\types\string.hpp"
#include "cereal\archives\portable_binary.hpp"
```
And my project explodes:
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	D9025	overriding '/sdl-' with '/GS-'	PeerNet	D:\Programming Backups\PeerNet\PeerNet\cl	1	
Error (active)		expected a declaration	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	285	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	302	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	314	
Error (active)		'constexpr' is not valid here	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	359	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	360	
Error (active)		'constexpr' is not valid here	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	781	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	781	
Error (active)		inline specifier allowed on function declarations only	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	1047	
Error (active)		namespace "chrono" has no member "duration"	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	1047	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	1047	
Error (active)		expected a declaration	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\chrono	1117	
Error (active)		variable "_Thrd_start" may not be initialized	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\thr\xthrcommon.h	81	
Error (active)		identifier "_Thrd_imp_t" is undefined	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\thr\xthrcommon.h	81	
Error (active)		expected an expression	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\thr\xthrcommon.h	81	
Error (active)		expected a declaration	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\thr\xthrcommon.h	95	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\tuple	959	
Error (active)		expected a ';'	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\tuple	998	
Error (active)		divides is not a template	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xfunctional	241	
Error (active)		expected a declaration	PeerNet	c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xfunctional	987	
Error (active)		identifier "clock_t" is undefined	PeerNet	c:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\time.h	137	
Error (active)		expected a declaration	PeerNet	c:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\time.h	600	
Error (active)		cannot open source file "cereal/cereal.hpp"	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	32	
Error (active)		OutputArchive is not a template	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	78	
Error (active)		identifier "AllowEmptyClassElision" is undefined	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	78	
Error (active)		InputArchive is not a template	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	178	
Error (active)		expected a ';'	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	285	
Error (active)		explicit type is missing ('int' assumed)	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	324	
Error (active)		explicit type is missing ('int' assumed)	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	325	
Error (active)		explicit type is missing ('int' assumed)	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\archives\portable_binary.hpp	328	
Error (active)		cannot open source file "cereal/cereal.hpp"	PeerNet	d:\Programming Backups\PeerNet\PeerNet\cereal\types\string.hpp	33	
Error (active)		expected a ';'	PeerNet	d:\Programming Backups\PeerNet\PeerNet\NetSocket.cpp	400	
Error (active)		"NetSocket" is not a type name	PeerNet	d:\Programming Backups\PeerNet\PeerNet\PeerNet.h	46	
Error (active)		incomplete type is not allowed	PeerNet	d:\Programming Backups\PeerNet\PeerNet\PeerNet.h	47	
Error (active)		type name is not allowed	PeerNet	d:\Programming Backups\PeerNet\PeerNet\PeerNet.h	47	
Error (active)		expected a ')'	PeerNet	d:\Programming Backups\PeerNet\PeerNet\PeerNet.h	47	
Warning	D9025	overriding '/sdl-' with '/GS-'	PeerNet	D:\Programming Backups\PeerNet\PeerNet\cl	1	
Error	C1083	Cannot open include file: 'cereal/cereal.hpp': No such file or directory	PeerNet	d:\programming backups\peernet\peernet\cereal\types\string.hpp	33	
Error	C1083	Cannot open include file: 'cereal/cereal.hpp': No such file or directory	PeerNet	d:\programming backups\peernet\peernet\cereal\types\string.hpp	33	
```
It seems to be due to the fact that cereal's files use <> in all their internal includes instead of ""
Why is this? I should be able to just drop the cereal folder inside my project folder and be good to go.

**Comment by kklouzal** [comment]
Yep, going through and manually changing ALL the `#include <cereal\*.hpp>"` to `#include "cereal\*.hpp"` remedied the issue.

It is to my understanding that by default per the C Standard section 6.10.2 paragraphs 2 and 3 state:

> A preprocessing directive of the form
> #include \<h-char-sequence\> new-line
> searches a sequence of implementation-defined places for a header identified uniquely by the specified sequence between the < and > delimiters, and causes the replacement of that directive by the entire contents of the header. How the places are specified or the header identified is implementation-defined.

> A preprocessing directive of the form
> #include "q-char-sequence" new-line
> causes the replacement of that directive by the entire contents of the source file identified by the specified sequence between the " delimiters. The named source file is searched for in an implementation-defined manner. If this search is not supported, or if the search fails, the directive is reprocessed as if it read
> #include <h-char-sequence> new-line
> with the identical contained sequence (including > characters, if any) from the original directive.

And that a common rule of thumb is `#include <some_header.h>` is reserved to include standard system libraries leaving `#include "some_header.h"` for the actual application/library files being programmed.

**Comment by AzothAmmo** [comment]
Did you try compiling with `-I /path/to/cereal`?

**Comment by kklouzal** [comment]
Yes, specifying a path to cereal when it resides outside your projects source/header folder works however you should be able to just drop the cereal folder into your projects source/header folder and be good to go.

Changing the internal cereal includes to use quotes instead of less than/greater than signs will not break any compatibility and not require anyone currently using the library to make any changes as using quotes defaults to using less than/greater than when the search path from using quotes cannot locate the specified file.

Using quotes just simplifies the inclusion of cereal for people that want drag and drop implementation. It also complies with the C Standard.

**Comment by pplux** [comment]
Same here, the library would be much easier to embed in a project if the headers were relative to each other, otherwise you need to change the compilation flags in order to make it work properly.

**Comment by AzothAmmo** [comment]
Merged #347, I need to do further testing on Windows but so far this seems like a reasonable change all around.
